core/translate/expr.rs

```diff
@@ -1615,6 +1629,66 @@ pub fn translate_expr(
                 }
                 Ok(target_register)
             }
+            (UnaryOperator::Negative, _) => {
+                let reg = program.alloc_register();
+                translate_expr(
+                    program,
+                    referenced_tables,
+                    expr,
+                    reg,
+                    precomputed_exprs_to_registers,
+                )?;
+                let zero_reg = program.alloc_register();
+                program.emit_insn(Insn::Integer {
```

Comment:
can you add `program.mark_last_insn_constant();` after this instruction? it basically stores the literal into a reg once at the beginning of the program instead of wasting cycles doing it in a loop.

also, could you do it for the Literals in the previous match arm too? I know you didn't write those parts but good to fix. 

I want to add an utility so that we don't have to remember to do this every time, but it's out of scope of this PR

Reply:
Sure thing :+1: . 

>I want to add an utility so that we don't have to remember to do this every time, but it's out of scope of this PR

 Maybe I can implement this in the next PR, I'll look into it.